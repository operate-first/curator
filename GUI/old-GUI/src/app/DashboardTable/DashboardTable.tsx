import React from 'react';
import { Table, TableHeader, TableBody, TableVariant, BodyCell } from '@patternfly/react-table';
import { Link } from 'react-router-dom';
import { dataObject } from '@app/project_page/demoProjectfilterform';

type DashboardTableProps = {
  columnTitle: object;
  tableData: Array<dataObject>;
};

type DashboardTableState = {
  columns: Array<object | string>;
  rows: Array<row>;
};

type row = {
  cells: cells;
};

type cells = Array<JSX.Element | number>;

class DashboardTable extends React.Component<DashboardTableProps, DashboardTableState> {
  constructor(myProps: DashboardTableProps) {
    super(myProps);

    const rowData: Array<row> = [];

    myProps.tableData.forEach(dataRow => {
      rowData.push({
        cells: [
          <BodyCell key={`/projectlist/${dataRow['namespace']}`}>
            <Link to={`/projectlist/${dataRow['namespace']}`} key={`/projectlist/${dataRow['namespace']}`}>
              {dataRow['namespace']}
            </Link>
          </BodyCell>
        ]
      });
    })

    this.state = {
      columns: [
        myProps.columnTitle['namespace']
      ],
      rows: rowData
    };
  }


  shouldComponentUpdate(nextProps: DashboardTableProps, nextState: DashboardTableState) {
    return JSON.stringify(nextProps.tableData) !== JSON.stringify(this.props.tableData)
  }

  UNSAFE_componentWillReceiveProps(nextProps: DashboardTableProps) {
    const rowData: Array<row> = [];

    nextProps.tableData.forEach(dataRow => {
      rowData.push({
        cells: [
          <BodyCell key={`/projectlist/${dataRow['namespace']}`}>
            <Link to={`/projectlist/${dataRow['namespace']}`} key={`/projectlist/${dataRow['namespace']}`}>
              {dataRow['namespace']}
            </Link>
          </BodyCell>
        ]
      });
    });
    this.setState({ ...this.state, rows: rowData });
  }

  render() {
    return (
      <div>
        <Table
          key={'dataTable'}
          aria-label="Compact Table"
          variant={TableVariant.compact}
          cells={this.state.columns}
          rows={this.state.rows}
          caption="List of projects"
        >
          {/* rowWrapper={} */}
          <TableHeader />
          <TableBody />
        </Table>
      </div>
    );
  }
}
export { DashboardTable };
