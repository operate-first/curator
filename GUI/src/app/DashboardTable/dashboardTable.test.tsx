import React from 'react';
import { shallow } from 'enzyme';
import { DashboardTable } from './DashboardTable';

describe('Dashboard tests', () => {
   
  // DashboardTable tests
  test('should render default DashboardTable component', () => {
    const testTableData = [{namespace: "openshift-monitoring", activationTime: 2},
    {namespace: "hello-world", activationTime: 1},
    {namespace: "myproject", activationTime: 2},
    {namespace: "sample-project", activationTime: 3},
    {namespace: "sample-project-1", activationTime: 1},
    {namespace: "sample-project-2", activationTime: 1},
    {namespace: "sample-project-3", activationTime: 1}]
    const testColumnTitle = {namespace: "Namespace", activationTime: "Project Active period"}
    const view = shallow(<DashboardTable tableData={testTableData} columnTitle={testColumnTitle}/>);
    expect(view).toMatchSnapshot();
  })

})