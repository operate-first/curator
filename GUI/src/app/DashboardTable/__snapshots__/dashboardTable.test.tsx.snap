// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard tests should render default DashboardTable component 1`] = `
<div>
  <Component
    aria-label="Compact Table"
    caption="List of projects"
    cells={
      Array [
        "Namespace",
        "Project Active period",
      ]
    }
    key="dataTable"
    rows={
      Array [
        Object {
          "cells": Array [
            <BodyCell>
              <Link
                to="/projectlist/openshift-monitoring"
              >
                openshift-monitoring
              </Link>
            </BodyCell>,
            2,
          ],
        },
        Object {
          "cells": Array [
            <BodyCell>
              <Link
                to="/projectlist/hello-world"
              >
                hello-world
              </Link>
            </BodyCell>,
            1,
          ],
        },
        Object {
          "cells": Array [
            <BodyCell>
              <Link
                to="/projectlist/myproject"
              >
                myproject
              </Link>
            </BodyCell>,
            2,
          ],
        },
        Object {
          "cells": Array [
            <BodyCell>
              <Link
                to="/projectlist/sample-project"
              >
                sample-project
              </Link>
            </BodyCell>,
            3,
          ],
        },
        Object {
          "cells": Array [
            <BodyCell>
              <Link
                to="/projectlist/sample-project-1"
              >
                sample-project-1
              </Link>
            </BodyCell>,
            1,
          ],
        },
        Object {
          "cells": Array [
            <BodyCell>
              <Link
                to="/projectlist/sample-project-2"
              >
                sample-project-2
              </Link>
            </BodyCell>,
            1,
          ],
        },
        Object {
          "cells": Array [
            <BodyCell>
              <Link
                to="/projectlist/sample-project-3"
              >
                sample-project-3
              </Link>
            </BodyCell>,
            1,
          ],
        },
      ]
    }
    variant="compact"
  >
    <TableHeader />
    <TableBody />
  </Component>
</div>
`;
